test/prism_regression/literal_integer.rb

```diff
@@ -7,3 +7,27 @@
 
 # ~ should not be a method call
 ~42
+
+# Hexadecimal literal
+0xcafe
```

Comment:
Can you also add tests for underscores in each of these? e.g. `1_234`, `0xBEER_CAFE`, `0x0000_1111`, `0o777_777`, etc.?